component_sdk/python/kfp_component/google/bigquery/_query.py

```diff
@@ -56,13 +80,41 @@ def _get_job(client, job_id):
     except exceptions.NotFound:
         return None
 
+def _prepare_dataset_ref(client, dataset_id, output_gcs_path):
+    if not output_gcs_path and not dataset_id:
+        return None
+    
+    if not dataset_id:
+        dataset_id = 'kfp_tmp_dataset'
+    dataset_ref = client.dataset(dataset_id)
+    dataset = _get_dataset(client, dataset_ref)
+    if not dataset:
+        logging.info('Creating dataset {}'.format(dataset_id))
+        dataset = _create_dataset(client, dataset_ref)
+    return dataset_ref
+
+def _get_dataset(client, dataset_ref):
+    try:
+        return client.get_dataset(dataset_ref)
+    except exceptions.NotFound:
+        return None
+
+def _create_dataset(client, dataset_ref):
+    dataset = bigquery.Dataset(dataset_ref)
+    dataset.location = "US"
+    return client.create_dataset(dataset)
+
 def _display_job_link(project_id, job_id):
     display.display(display.Link(
         href= 'https://console.cloud.google.com/bigquery?project={}'
             '&j={}&page=queryresults'.format(project_id, job_id),
         text='Query Details'
     ))
 
-def _dump_outputs(job):
-    gcp_common.dump_file('/tmp/outputs/bigquery-job.json', 
-        json.dumps(job.to_api_repr()))
\ No newline at end of file
+def _dump_outputs(job, output_path):
+    gcp_common.dump_file('/tmp/kfp/output/biquery/query-job.json', 
```

Comment:
It would be best if we did not hardcode any paths inside the library. It's considered bad practice in all programming languages including Python.

https://en.wikipedia.org/wiki/Hard_coding
https://en.wikipedia.org/wiki/Anti-pattern

Reply:
The output folder will be replaced by an environment variable in the future. I will add a TODO for it.